edgexfoundry/edgex-go
# feat(metadata): Implement Writable.ProfileChange configurations

Signed-off-by: Ginny Guan <ginny@iotechsys.com>







**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  <link to docs PR>
https://github.com/edgexfoundry/edgex-docs/pull/727
Also update metadata Swagger file in this PR

## Testing Instructions

1. build core-metadata from this branch
2. set Writable.ProfileChange settings to false and run core-metadata
3. verify `PUT /deivceprofile`, `DELETE /deivceprofile`, `PUT /deviceprofile/upload`, and `PATCH /deviceprofile/basicinfo` responses
## New Dependency Instructions (If applicable)
